tgui-next/packages/tgui/interfaces/ScannerGate.js

```diff
@@ -0,0 +1,330 @@
+      />
+      {!data.locked && (
+        <ScannerGateControl state={state} />
+      )}
+    </Fragment>
+  );
+};
+
+const SCANNER_GATE_ROUTES = {
+  Off: {
+    title: 'Scanner Mode: Off',
+    component: () => ScannerGateOff,
+  },
+  Wanted: {
+    title: 'Scanner Mode: Wanted',
+    component: () => ScannerGateWanted,
+  },
+  Guns: {
+    title: 'Scanner Mode: Guns',
+    component: () => ScannerGateGuns,
+  },
+  Mindshield: {
+    title: 'Scanner Mode: Mindshield',
+    component: () => ScannerGateMindshield,
+  },
+  Disease: {
+    title: 'Scanner Mode: Disease',
+    component: () => ScannerGateDisease,
+  },
+  Species: {
+    title: 'Scanner Mode: Species',
+    component: () => ScannerGateSpecies,
+  },
+  Nutrition: {
+    title: 'Scanner Mode: Nutrition',
+    component: () => ScannerGateNutrition,
+  },
+  Nanites: {
+    title: 'Scanner Mode: Nanites',
+    component: () => ScannerGateNanites,
+  },
+};
+
+const ScannerGateControl = props => {
+  const { state } = props;
+  const { config, data } = state;
+  const { ref } = config;
+  const route = SCANNER_GATE_ROUTES[data.scan_mode] || SCANNER_GATE_ROUTES.off;
+  const Component = route.component();
+
+  return (
+    <Section
+      title={route.title}
+      buttons={data.scan_mode !== 'Off' && (
+        <Button
+          icon="arrow-left"
+          content="back"
+          onClick={() => act(ref, 'set_mode', { new_mode: "Off" })}
+        />
+      )}>
+      <Component state={state} />
+    </Section>
+  );
+};
+
+const ScannerGateOff = props => {
+  const { state } = props;
+  const { config } = state;
+  const { ref } = config;
+
+  return (
+    <Fragment>
+      <Box mb={2}>
+        Select a scanning mode below.
+      </Box>
+      <Box>
+        <Button
+          content="Wanted"
+          onClick={() => act(ref, 'set_mode', { new_mode: "Wanted" })}
+        />
+        <Button
+          content="Guns"
+          onClick={() => act(ref, 'set_mode', { new_mode: "Guns" })}
+        />
+        <Button
+          content="Mindshield"
+          onClick={() => act(ref, 'set_mode', { new_mode: "Mindshield" })}
+        />
+        <Button
+          content="Disease"
+          onClick={() => act(ref, 'set_mode', { new_mode: "Disease" })}
+        />
+        <Button
+          content="Species"
+          onClick={() => act(ref, 'set_mode', { new_mode: "Species" })}
+        />
+        <Button
+          content="Nutrition"
+          onClick={() => act(ref, 'set_mode', { new_mode: "Nutrition" })}
+        />
+        <Button
+          content="Nanites"
+          onClick={() => act(ref, 'set_mode', { new_mode: "Nanites" })}
+        />
+      </Box>
+    </Fragment>
+  );
+};
+
+const ScannerGateWanted = props => {
+  const { state } = props;
+  const { data } = state;
+
+  return (
+    <Fragment>
+      <Box mb={2}>
+        Person&#39;s scanned that {data.reverse ? "do not have" : "has"} any warrants for their arrest
+        will trigger the gate.
```

Comment:
Persons

This entire sentence doesn't really roll off the tongue very easily. Maybe "Triggers if scanned entity [has/is thing]"?
Same for the rest of them.

Reply:
Person's is used because it does not scan all entities, only common people, abductors, shades, etc do not get scanned and they are entities.
Per the wiki and code.

Reply:
"Person's scanned" -> just "Person"

Reply:
Person with nanite cloud X...

Or what about a Carbon lifeform?

Reply:
Right, but all the scanned things are entities :nerd_face: Up to you what you call them, but I would still argue for changing up the sentence structure.

Reply:
Wording and changes to species selection incoming.
I did not like the name of zombie, had to do work to fix it up.